starboard/nplb/player_get_audio_configuration_test.cc

```diff
@@ -0,0 +1,198 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "starboard/nplb/player_test_fixture.h"
+#include "starboard/string.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+#if SB_API_VERSION >= SB_MEDIA_ENHANCED_AUDIO_API_VERSION
+
+bool operator==(const SbMediaAudioConfiguration& left,
+                const SbMediaAudioConfiguration& right) {
+  return memcmp(&left, &right, sizeof(SbMediaAudioConfiguration)) == 0;
+}
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+using ::testing::ValuesIn;
+
+typedef SbPlayerTestFixture::GroupedSamples GroupedSamples;
+
+#if SB_API_VERSION >= SB_MEDIA_ENHANCED_AUDIO_API_VERSION
+
+class SbPlayerGetAudioConfigurationTest
+    : public ::testing::TestWithParam<SbPlayerTestConfig> {
+ public:
+  void ReadAudioConfigurations(
+      const SbPlayerTestFixture& player_fixture,
+      std::vector<SbMediaAudioConfiguration>* configurations) {
+    const int kMaxNumberOfConfigurations = 32;
+    for (int index = 0;; index++) {
+      ASSERT_LT(index, kMaxNumberOfConfigurations);
+      SbMediaAudioConfiguration configuration;
+      if (SbPlayerGetAudioConfiguration(player_fixture.GetPlayer(), index,
+                                        &configuration)) {
+        configurations->push_back(configuration);
+      } else {
+        return;
+      }
+    }
+  }
+};
+
+TEST_P(SbPlayerGetAudioConfigurationTest, SunnyDay) {
+  const int kSamplesToWrite = 8;
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+
+  // It's ok to not return any configuration at this point.
+  std::vector<SbMediaAudioConfiguration> initial_configs;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &initial_configs));
+
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    samples.AddAudioSamplesWithEOS(0, kSamplesToWrite);
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamplesWithEOS(0, kSamplesToWrite);
+  }
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  // The audio configurations must be available after `kSbPlayerStatePresenting`
+  // has been received.
+  std::vector<SbMediaAudioConfiguration> configs_after_presenting;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_presenting));
+
+  if (player_fixture.HasAudio()) {
+    ASSERT_FALSE(configs_after_presenting.empty());
+  }
+
+  // Once at least one audio configurations are returned, the return values
```

Comment:
Let's refine the comment of `SbPlayerGetAudioConfiguration()` to explicitly mention that the order shouldn't change during the life time too.